ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/user-notifications/user-notifications.view.js

```diff
@@ -0,0 +1,189 @@
+var user = require('../../../component/singletons/user-instance.js')
+var moment = require('moment')
+var userNotifications = require('../../../component/singletons/user-notifications.js')
+
+const Empty = styled.div`
+  transition: transform ${props => props.theme.coreTransitionTime} linear;
+  transform: scale(1);
+  text-align: center;
+  font-size: ${props => props.theme.largeFontSize}
+  padding: ${props => props.theme.mediumSpacing}
+`
+const Notifications = styled.div`
+  height: 100%;
+  width: 100%;
+  display: block;
+  overflow: auto;
+  padding: ${props => props.theme.mediumSpacing};
+`
+
+class UserNotifications extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = userNotifications
+    this.props.listenTo(
+      userNotifications,
+      'add remove update',
+      this.updateState.bind(this)
+    )
+  }
+  render() {
+    return userNotifications.isEmpty() ? (
+      <Empty>
+        <div className="notifications-empty">No Notifications</div>
+      </Empty>
+    ) : (
+      <Notifications>
+        <MarionetteRegionContainer
+          classname="list-today"
+          view={this.listToday()}
+          viewOptions={this.replaceElement()}
+        />
+        <MarionetteRegionContainer
+          classname="list-one"
+          view={this.listPreviousOne()}
+          viewOptions={this.replaceElement()}
+        />
+        <MarionetteRegionContainer
+          classname="list-two"
+          view={this.listPreviousTwo()}
+          viewOptions={this.replaceElement()}
+        />
+        <MarionetteRegionContainer
+          classname="list-three"
+          view={this.listPreviousThree()}
+          viewOptions={this.replaceElement()}
+        />
+        <MarionetteRegionContainer
+          classname="list-four"
+          view={this.listPreviousFour()}
+          viewOptions={this.replaceElement()}
+        />
+        <MarionetteRegionContainer
+          classname="list-five"
+          view={this.listPreviousFive()}
+          viewOptions={this.replaceElement()}
+        />
+        <MarionetteRegionContainer
+          classname="list-six"
+          view={this.listPreviousSix()}
+          viewOptions={this.replaceElement()}
+        />
+        <MarionetteRegionContainer
+          classname="list-older"
+          view={this.listOlder()}
+          viewOptions={this.replaceElement()}
+        />
+      </Notifications>
+    )
+  }
+  updateState() {
+    this.setState(userNotifications)
+  }
+  listToday() {
+    return new NotificationGroupView({
+      filter: function(model) {
+        return moment().diff(model.get('sentAt'), 'days') === 0
+      },
+      date: 'Today',
+    })
+  }
+  listPreviousOne() {
+    return new NotificationGroupView({
+      filter: function(model) {
+        return moment().diff(model.get('sentAt'), 'days') === 1
+      },
+      date: 'Yesterday',
+    })
+  }
+  listPreviousTwo() {
```

Comment:
consider replacing these functions with a single function that takes the number of days as an argument. replace calls to this function w/ `listPrevious(2)` etc

Reply:
(and isolating the logic that returns `Yesterday`, `Older`, or the moment result into a `informalName(daysAgo)` function)